tendermint/src/lite_impl/signed_header.rs

```diff
@@ -47,6 +47,12 @@ impl lite::Commit for block::signed_header::SignedHeader {
         if self.commit.precommits.len() != vals.validators().len() {
             return Err(lite::Error::InvalidCommitSignatures);
         }
+        if let Some(parts) = &self.commit.block_id.parts {
+            if parts.hash != self.header_hash() {
```

Comment:
@Liamsi `parts.hash` is the `merkleRoot(partsHeader)` which is different from the `header_hash`. That's why this check is making many other tests to fail too. We want to check that the `commit.BlockID` is equal to each of the `precommit.BlockID`. In reality, there can be precommits for other blocks and that's totally fine. But then, those precommits should not be considered for this block. This means that when we calculate the total voting power i.e. `signed_power`, we do not count voting power of validators who signed these precommits. Maybe this check should go under the `voting_power_in` method(?)

Reply:
I see. Thanks for the clarification! To move forward, I'd suggest to merge this, and open an issue referencing this comment and the failing test.

Reply:
Oh, is this about the bug I raised here: https://github.com/interchainio/tendermint-rs/issues/145#issuecomment-583790278 ? We should be testing (either here or in voting_power_in) that the votes being tallied are for the right block ID